app/controllers/course/experience_points_records_controller.rb

```diff
@@ -8,28 +8,34 @@ class Course::ExperiencePointsRecordsController < Course::ComponentController
   before_action :add_breadcrumbs
 
   def index # :nodoc:
-    updater_ids = @experience_points_records.active.pluck(:updater_id)
-    @course_user_preload_service =
-      Course::CourseUserPreloadService.new(updater_ids, current_course)
-
-    @experience_points_records =
-      @experience_points_records.active.
-      includes(:actable, :updater).order(updated_at: :desc).page(page_param)
+    respond_to do |format|
+      format.html
+      format.json do
+        updater_ids = @experience_points_records.active.pluck(:updater_id)
+        @course_user_preload_service =
+          Course::CourseUserPreloadService.new(updater_ids, current_course)
+        @experience_points_records =
+          @experience_points_records.active.
+          includes(:actable, :updater).order(updated_at: :desc)
+        @experience_points_count = @experience_points_records.count
+        @experience_points_records = @experience_points_records.paginated(paginate_page_param)
+      end
+    end
   end
 
   def update
     if @experience_points_record.update(experience_points_record_params)
-      flash.now[:success] = t('.success')
+      head :ok
     else
-      flash.now[:danger] = @experience_points_record.errors.full_messages.to_sentence
+      head :bad_request
     end
   end
 
   def destroy # :nodoc:
     if @experience_points_record.destroy
-      destroy_success
+      head :ok
     else
-      destroy_failure
+      head :bad_request
```

Comment:
Pass `@experience_points_record.errors.full_messages.to_sentence` to frontend and render it as well using toast.